torch/distributed/device_mesh.py

```diff
@@ -477,6 +428,20 @@ def __init__(
             self.mesh_dim_names = tuple(mesh_dim_names) if mesh_dim_names else None
             if backend_override is None:
                 backend_override = ((None, None),) * self.mesh.ndim
+            # Internal bookkeeping for the device mesh.
+            self._layout = (
+                _layout
+                if _layout
+                else _MeshLayout(self.mesh.size(), self.mesh.stride())
```

Comment:
Does the layout always have to be consistent with self.mesh.size()/stride()?

Reply:
Its numel has to be the same, but the value of size and stride no. That's exactly why we need to scale in the remapping.

Reply:
So in an ideal world, layout always has to be consistent with self.mesh.size()/stride(). But currently we have couple constraints:
1. All ranks need to call backend init, even if that ranks are not part it. So we will want to keep the layout to be same as before the submesh is sliced. This is also why we need to do scale.
2. Another thing we need to support slicing out flattened mesh from root mesh.

We believe both will be gone eventually and we will be able to eventually come back to this consistency and get rid of scaling as well.